tests/unit/s2n_random_test.c

```diff
@@ -481,6 +546,17 @@ static int s2n_common_tests(struct random_test_case *test_case)
     EXPECT_OK(s2n_fork_test(s2n_get_public_random_data, s2n_get_private_random_data));
     EXPECT_OK(s2n_fork_test(s2n_get_private_random_data, s2n_get_public_random_data));
 
+    /* Some fork detection mechanisms can also detect forks through clone() */
+    if (test_case->test_case_must_pass_clone_test == CLONE_TEST_YES) {
```

Comment:
I'm not really following why this 'test_case_must_pass_clone_test' parameter is necessary. Why can't you just do 
```
if (s2n_is_madv_wipeonfork_supported() || s2n_is_map_inherit_zero_supported()) {
            EXPECT_OK(s2n_execute_clone_tests());
}
```

Reply:
I'm not following how this makes the test stronger. What are we losing by taking it out?

Reply:
Each test case runs with a set of fork detection methods active. For singleton tests (with only one specific fork detection method chosen), I know whichtest cases that should be able to handle detecting forks through a `clone()` operation. So, I mark such test cases with that information. 

For some test cases, I only know later on whether the active fork detection methods can handle detecting forks through `clone()`.

That's what I have tried to capture in the labels and I handle this in if-else logic.

But, I think you are right in that the entire logic might just boil down to the else-branch in the end. And, as you point out, this actually doesn't bring any value anyway.